Core/include/Acts/TrackFitting/detail/KalmanUpdateHelpers.hpp

```diff
@@ -59,9 +60,6 @@ auto kalmanHandleMeasurement(
 
   trackStateProxy.setReferenceSurface(surface.getSharedPtr());
 
-  // assign the source link to the track state
-  trackStateProxy.setUncalibratedSourceLink(source_link);
```

Comment:
not needed anymore?

Reply:
The idea is that the calibrator does this now. For the refitting case it can then unpack

SourceLink -> TrackStateProxy -> original SourceLink.

See `RefittingCalibrator.hpp`